compiler/damlc/daml-opts/daml-opts-types/DA/Daml/Options/Types.hs

```diff
@@ -116,11 +116,14 @@ mkOptions opts@Options {..} = do
           unless ok $ fail $ "Required directory does not exist: " <> f
         versionSuffix = renderPretty optDamlLfVersion
 
--- | Default configuration for the compiler with package database set according to daml-lf version
--- and located runfiles. If the version argument is Nothing it is set to the default daml-lf
--- version.
+-- | Default configuration for the compiler with package database set
+-- according to daml-lf version and located runfiles. If the version
+-- argument is Nothing it is set to the default daml-lf
+-- version. Linting is enabled but not '.dlint.yaml' overrides.
 defaultOptionsIO :: Maybe LF.Version -> IO Options
-defaultOptionsIO mbVersion = mkOptions $ defaultOptions mbVersion
+defaultOptionsIO mbVersion = do
+  hlintDataDir <-locateRunfiles $ mainWorkspace </> "compiler/damlc/daml-ide-core"
```

Comment:
Should this maybe go into `mkOptions`? That seems to match how we handle things like the package db.